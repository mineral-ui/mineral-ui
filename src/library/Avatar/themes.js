/* @flow */
import type { AvatarThemeFn } from './types';

export const avatarTheme: AvatarThemeFn = (baseTheme) => ({
  Avatar_fontSize_small: baseTheme.fontSize_mouse,
  Avatar_fontSize_medium: baseTheme.fontSize_ui,
  Avatar_fontSize_large: baseTheme.h4_fontSize,
  Avatar_fontSize_jumbo: baseTheme.h4_fontSize,
  Avatar_fontWeight: baseTheme.fontWeight_bold,
  Avatar_size_small: baseTheme.size_small,
  Avatar_size_medium: baseTheme.size_medium,
  Avatar_size_large: baseTheme.size_large,
  Avatar_size_jumbo: baseTheme.size_jumbo,

  ...baseTheme
});
