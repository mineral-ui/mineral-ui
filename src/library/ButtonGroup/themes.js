/* @flow */
import type { ButtonGroupThemeFn } from './types';

export const buttonGroupTheme: ButtonGroupThemeFn = (baseTheme) => ({
  ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
  ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
  ButtonGroupButton_borderStartColor: baseTheme.borderColor,
  ButtonGroupButton_borderStartColor_checked: 'currentcolor',
  ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,

  ...baseTheme
});
