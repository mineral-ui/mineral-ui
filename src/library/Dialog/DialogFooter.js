/* @flow */
import React from 'react';
import { createStyledComponent } from '../styles';
import DialogRow from './DialogRow';

type Props = {
  /** TODO */
  children: React$Node
};

export const componentTheme = (baseTheme: Object) => ({
  DialogFooterItem_margin: baseTheme.space_stack_md,

  ...baseTheme
});

const styles = {
  root: ({ theme: baseTheme }) => {
    const theme = componentTheme(baseTheme);
    const rtl = theme.direction === 'rtl';

    return {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'flex-end',

      '& > *:not(:last-child)': {
        marginLeft: rtl ? theme.DialogFooterItem_margin : null,
        marginRight: rtl ? null : theme.DialogFooterItem_margin
      }
    };
  }
};

const Root = createStyledComponent(DialogRow, styles.root, {
  displayName: 'DialogFooter',
  withProps: { element: 'footer' }
});

/**
 * DialogFooter - TODO
 */
export default function DialogFooter(props: Props) {
  return <Root {...props} />;
}
