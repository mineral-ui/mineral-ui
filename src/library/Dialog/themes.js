/* @flow */
import { pxToEm } from '../styles';
import { mapComponentThemes } from '../themes';
import { overflowContainerWithShadowsTheme } from '../OverflowContainer/themes';

import type {
  DialogThemeFn,
  DialogActionsThemeFn,
  DialogBodyThemeFn,
  DialogRowThemeFn,
  DialogTitleThemeFn
} from './types';

export const dialogTheme: DialogThemeFn = (baseTheme) => ({
  Dialog_transitionDuration: '250ms',
  Dialog_zIndex: baseTheme.zIndex_1600,

  DialogCloseButton_margin: baseTheme.space_inline_sm,

  DialogContent_backgroundColor: baseTheme.panel_backgroundColor,
  DialogContent_borderColor: baseTheme.panel_borderColor,
  DialogContent_borderRadius: baseTheme.borderRadius_1,
  DialogContent_boxShadow: baseTheme.boxShadow_5,
  DialogContent_maxHeight: '80vh',
  DialogContent_maxHeight_small: pxToEm(560),
  DialogContent_maxHeight_medium: pxToEm(560),
  DialogContent_maxHeight_large: pxToEm(720),
  DialogContent_maxWidth_small: pxToEm(400),
  DialogContent_maxWidth_medium: pxToEm(640),
  DialogContent_maxWidth_large: pxToEm(1200),
  DialogContent_minWidth: pxToEm(360),
  DialogContent_offsetVertical: baseTheme.space_stack_xxl,
  DialogContent_width_small: '35vw',
  DialogContent_width_medium: '50vw',
  DialogContent_width_large: '80vw',

  DialogOverlay_backgroundColor: 'rgba(0, 0, 0, 0.6)',

  ...dialogRowTheme(baseTheme),
  ...baseTheme
});

export const dialogActionsTheme: DialogActionsThemeFn = (baseTheme) => ({
  DialogActionsItem_margin: baseTheme.space_stack_sm,

  ...baseTheme
});

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const dialogBodyTheme: DialogBodyThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'OverflowContainerWithShadows',
      theme: overflowContainerWithShadowsTheme(baseTheme)
    },
    {
      name: 'DialogBody',
      theme: {}
    },
    baseTheme
  );

export const dialogRowTheme: DialogRowThemeFn = (baseTheme) => ({
  DialogRow_fontSize: baseTheme.fontSize_ui,
  DialogRow_paddingHorizontal: baseTheme.space_inline_lg,
  DialogRow_marginVertical: baseTheme.space_stack_lg,

  ...baseTheme
});

export const dialogTitleTheme: DialogTitleThemeFn = (baseTheme) => ({
  DialogTitle_color: baseTheme.h4_color,
  DialogTitle_fontSize: baseTheme.h4_fontSize,
  DialogTitle_fontWeight: baseTheme.h4_fontWeight,

  DialogTitleIcon_margin: baseTheme.space_inline_sm,
  DialogTitleIcon_size: pxToEm(24),

  ...baseTheme
});
