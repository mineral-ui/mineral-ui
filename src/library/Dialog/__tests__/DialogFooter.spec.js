/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import { dialogRowTheme } from '../themes';
import DialogFooter from '../DialogFooter';
import examples from '../../../website/app/demos/Dialog/DialogFooter/examples';
import testDemoExamples from '../../../../utils/testDemoExamples';
import testThemeOverrides from '../../../../utils/testThemeOverrides';
import { getProcessedComponentThemeKeys } from '../../themes/processComponentTheme';

function shallowDialogFooter(props = {}) {
  const dialogFooterProps = {
    ...props
  };

  return shallow(<DialogFooter {...dialogFooterProps} />);
}

describe('DialogFooter', () => {
  it('renders', () => {
    const dialogFooter = shallowDialogFooter();

    expect(dialogFooter.exists()).toEqual(true);
  });

  testDemoExamples(examples);

  describe('theme overrides', () => {
    testThemeOverrides(
      <DialogFooter />,
      getProcessedComponentThemeKeys(dialogRowTheme)
    );
  });
});
