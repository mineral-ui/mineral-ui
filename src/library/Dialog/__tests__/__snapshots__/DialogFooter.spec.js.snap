// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogFooter demo examples Snapshots: basic 1`] = `
.emotion-30 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-29 {
  box-sizing: border-box;
  color: #333840;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.875em;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  margin: 1.7142857142857142em 0;
  padding: 0 1.7142857142857142em;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-29 *,
.emotion-29 *::before,
.emotion-29 *::after {
  box-sizing: inherit;
}

.emotion-38 {
  padding: 1em;
  position: relative;
}

.emotion-38:not(:last-child) {
  padding-bottom: 0;
}

.emotion-38 > div {
  position: static;
}

.emotion-38 [role="document"] {
  width: auto;
}

.emotion-35 {
  position: static;
}

.emotion-36 {
  opacity: 0;
  position: relative;
  -webkit-transition: opacity 250ms ease;
  transition: opacity 250ms ease;
  will-change: opacity;
  z-index: 1600;
}

.emotion-34 {
  box-sizing: border-box;
  color: #333840;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  position: static;
}

.emotion-34 *,
.emotion-34 *::before,
.emotion-34 *::after {
  box-sizing: inherit;
}

.emotion-0 {
  background-color: rgba(0,0,0,0.6);
  bottom: 0;
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
}

.emotion-33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-32 {
  background-color: #ffffff;
  border: 1px solid #ebeff5;
  border-radius: 0.1875em;
  box-shadow: 0 16px 24px 0 rgba(0,0,0,0.2),0 32px 24px -16px rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 80vh;
  min-width: 22.5em;
  pointer-events: all;
  position: relative;
  max-width: 40em;
  width: 50vw;
}

@media (min-height:43em) {
  .emotion-32 {
    max-height: 35em;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-6 {
  box-sizing: border-box;
  color: #333840;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.875em;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  margin: 1.7142857142857142em 0;
  padding: 0 1.7142857142857142em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-6 *,
.emotion-6 *::before,
.emotion-6 *::after {
  box-sizing: inherit;
}

.emotion-5 {
  color: #58606e;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 > [role="img"] {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 0.5em;
}

.emotion-2 {
  color: inherit;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-1 {
  box-sizing: border-box;
  color: #58606e;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 18px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  font-weight: 700;
  margin-bottom: 0;
  margin-top: 0;
  color: inherit;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-1 *,
.emotion-1 *::before,
.emotion-1 *::after {
  box-sizing: inherit;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 0.875em;
  margin: 0;
  min-height: 0%;
  padding: 0;
}

.emotion-20:first-child {
  margin-top: 1.7142857142857142em;
}

.emotion-20:last-child {
  margin-bottom: 1.7142857142857142em;
}

.emotion-19 {
  box-sizing: border-box;
  color: #333840;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.875em;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  margin: 1.7142857142857142em 0;
  padding: 0 1.7142857142857142em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 0.875em;
  margin: 0;
  min-height: 0%;
  padding: 0;
}

.emotion-19 *,
.emotion-19 *::before,
.emotion-19 *::after {
  box-sizing: inherit;
}

.emotion-19:first-child {
  margin-top: 1.7142857142857142em;
}

.emotion-19:last-child {
  margin-bottom: 1.7142857142857142em;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-15 > div {
  padding-left: 1.7142857142857142em;
  padding-right: 1.7142857142857142em;
}

.emotion-15 > div > :first-child {
  margin-top: 0;
}

.emotion-15 > div > :last-child {
  margin-bottom: 0;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-14 > div {
  padding-left: 1.7142857142857142em;
  padding-right: 1.7142857142857142em;
}

.emotion-14 > div > :first-child {
  margin-top: 0;
}

.emotion-14 > div > :last-child {
  margin-bottom: 0;
}

.emotion-10 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-x: hidden;
  padding-bottom: 2px;
  padding-top: 2px;
}

.emotion-9 {
  outline: 0;
  overflow-y: auto;
  -webkit-transition: outline 0.1s 0.25s;
  transition: outline 0.1s 0.25s;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-x: hidden;
  padding-bottom: 2px;
  padding-top: 2px;
}

.emotion-9:focus {
  outline: 1px solid #1d5bbf;
}

.emotion-28 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-28 > *:not(:last-child) {
  margin-right: 0.5em;
}

.emotion-24 {
  box-sizing: border-box;
  color: #3272d9;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: transparent;
  border-color: transparent;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: inline-block;
  font-weight: 600;
  height: 2em;
  margin: 0;
  min-width: 2em;
  padding: 0 0.5em;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
}

.emotion-24 *,
.emotion-24 *::before,
.emotion-24 *::after {
  box-sizing: inherit;
}

.emotion-24:focus {
  border-color: #1d5bbf;
  color: #3272d9;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-24:hover {
  background-color: #f5f7fa;
  color: #3272d9;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-24:focus:active,
.emotion-24:focus:hover {
  border-color: #1d5bbf;
}

.emotion-24:active {
  background-color: #ebeff5;
  color: #3272d9;
}

.emotion-24::-moz-focus-inner {
  border: 0;
}

.emotion-24 [role="img"] {
  box-sizing: content-box;
  color: #3272d9;
  display: block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-24 [role="img"]:first-child {
  margin-right: 0.5em;
}

.emotion-24 [role="img"]:last-child {
  margin-left: 0.5em;
}

.emotion-24 [role="img"]:only-child {
  margin: 0;
}

.emotion-23 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-height: 100%;
  pointer-events: none;
  width: 100%;
}

.emotion-22 {
  display: block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  font-size: 0.875em;
  line-height: 2.2857142857142856em;
}

.emotion-27 {
  box-sizing: border-box;
  color: #ffffff;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: #3272d9;
  border-color: transparent;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: inline-block;
  font-weight: 600;
  height: 2em;
  margin: 0;
  min-width: 2em;
  padding: 0 0.5em;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
}

.emotion-27 *,
.emotion-27 *::before,
.emotion-27 *::after {
  box-sizing: inherit;
}

.emotion-27:focus {
  background-color: #3272d9;
  box-shadow: 0 0 0 1px #ffffff,0 0 0 2px #1d5bbf;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-27:hover {
  background-color: #5691f0;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-27:focus:active,
.emotion-27:focus:hover {
  border-color: transparent;
}

.emotion-27:active {
  background-color: #1d5bbf;
  border-color: #1d5bbf;
  color: #ffffff;
}

.emotion-27::-moz-focus-inner {
  border: 0;
}

.emotion-27 [role="img"] {
  box-sizing: content-box;
  display: block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-27 [role="img"]:first-child {
  margin-right: 0.5em;
}

.emotion-27 [role="img"]:last-child {
  margin-left: 0.5em;
}

.emotion-27 [role="img"]:only-child {
  margin: 0;
}

<Styled(div)>
  <div
    className="emotion-38"
  >
    <withProps(DemoDialog)>
      <DemoDialog
        closeButtonLabel="close"
        closeOnClickOutside={false}
        closeOnEscape={false}
        disableFocusOnOpen={true}
        disableFocusTrap={true}
        isOpen={true}
        preventCloseButtonClose={true}
        size="medium"
        usePortal={false}
      >
        <DemoDialog
          className="emotion-35"
          closeButtonLabel="close"
          closeOnClickOutside={false}
          closeOnEscape={false}
          disableFocusOnOpen={true}
          disableFocusTrap={true}
          isOpen={true}
          preventCloseButtonClose={true}
          size="medium"
          usePortal={false}
        >
          <WithTheme(Component)
            in={true}
            onEntered={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
          >
            <Component
              in={true}
              onEntered={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
            >
              <Transition
                appear={true}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={true}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={250}
                unmountOnExit={true}
              >
                <Animate
                  state="entering"
                >
                  <div
                    className="emotion-36"
                  >
                    <FocusTrap
                      _createFocusTrap={[Function]}
                      active={false}
                      focusTrapOptions={Object {}}
                      paused={false}
                      tag="div"
                    >
                      <div>
                        <Dialog
                          aria-labelledby="dialog-1-title"
                          aria-modal={true}
                          className="emotion-35"
                          closeButtonLabel="close"
                          disableFocusOnOpen={true}
                          id="dialog-1"
                          innerRef={[Function]}
                          preventCloseButtonClose={true}
                          role="dialog"
                          tabIndex="-1"
                        >
                          <div
                            aria-labelledby="dialog-1-title"
                            aria-modal={true}
                            className="emotion-34"
                            id="dialog-1"
                            role="dialog"
                            tabIndex="-1"
                          >
                            <Overlay>
                              <div
                                className="emotion-0"
                              />
                            </Overlay>
                            <Styled(div)>
                              <div
                                className="emotion-33"
                              >
                                <DialogContent
                                  id="dialog-1-content"
                                  innerRef={[Function]}
                                  role="document"
                                  size="medium"
                                >
                                  <div
                                    className="emotion-32"
                                    id="dialog-1-content"
                                    role="document"
                                  >
                                    <DialogHeader
                                      titleProps={
                                        Object {
                                          "id": "dialog-1-title",
                                          "variant": undefined,
                                        }
                                      }
                                    >
                                      <withProps(DialogHeader)>
                                        <DialogHeader
                                          element="header"
                                        >
                                          <DialogHeader
                                            className="emotion-7"
                                            element="header"
                                          >
                                            <Styled(header)
                                              className="emotion-7"
                                              element="header"
                                            >
                                              <header
                                                className="emotion-6"
                                              >
                                                <WithTheme(DialogTitle)
                                                  id="dialog-1-title"
                                                >
                                                  <DialogTitle
                                                    appearance="h4"
                                                    element="h1"
                                                    id="dialog-1-title"
                                                  >
                                                    <DialogTitle
                                                      appearance="h4"
                                                      element="h1"
                                                    >
                                                      <div
                                                        className="emotion-5"
                                                      >
                                                        <TitleContent
                                                          appearance="h4"
                                                          element="h1"
                                                          noMargins={true}
                                                        >
                                                          <Styled(Text)
                                                            appearance="h4"
                                                            element="h1"
                                                            id="dialog-1-title"
                                                            noMargins={true}
                                                          >
                                                            <Text
                                                              appearance="h4"
                                                              className="emotion-2"
                                                              element="h1"
                                                              id="dialog-1-title"
                                                              noMargins={true}
                                                            >
                                                              <TextProvider
                                                                appearance="h4"
                                                                className="emotion-2"
                                                                element="h1"
                                                                id="dialog-1-title"
                                                                noMargins={true}
                                                              >
                                                                <Styled(h1)
                                                                  appearance="h4"
                                                                  className="emotion-2"
                                                                  element="h1"
                                                                  id="dialog-1-title"
                                                                  noMargins={true}
                                                                >
                                                                  <h1
                                                                    className="emotion-1"
                                                                    id="dialog-1-title"
                                                                  >
                                                                    Title
                                                                  </h1>
                                                                </Styled(h1)>
                                                              </TextProvider>
                                                            </Text>
                                                          </Styled(Text)>
                                                        </TitleContent>
                                                      </div>
                                                    </DialogTitle>
                                                  </DialogTitle>
                                                </WithTheme(DialogTitle)>
                                              </header>
                                            </Styled(header)>
                                          </DialogHeader>
                                        </DialogHeader>
                                      </withProps(DialogHeader)>
                                    </DialogHeader>
                                    <DialogBody>
                                      <DialogBody
                                        element="div"
                                      >
                                        <DialogHeader
                                          className="emotion-20"
                                          element="div"
                                        >
                                          <Styled(div)
                                            className="emotion-20"
                                            element="div"
                                          >
                                            <div
                                              className="emotion-19"
                                            >
                                              <withProps(Styled(WithTheme(Themed(OverflowContainerWithShadows))))>
                                                <Styled(WithTheme(Themed(OverflowContainerWithShadows)))
                                                  scrollY={true}
                                                >
                                                  <WithTheme(Themed(OverflowContainerWithShadows))
                                                    className="emotion-15"
                                                    scrollY={true}
                                                  >
                                                    <Themed(OverflowContainerWithShadows)
                                                      className="emotion-15"
                                                      scrollY={true}
                                                    >
                                                      <ThemeProvider>
                                                        <ThemeProvider>
                                                          <OverflowContainerWithShadows
                                                            className="emotion-15"
                                                            scrollY={true}
                                                          >
                                                            <OverflowContainerWithShadows
                                                              className="emotion-15"
                                                              hasShadowBottom={false}
                                                              hasShadowTop={false}
                                                            >
                                                              <div
                                                                className="emotion-14"
                                                              >
                                                                <Scroller
                                                                  containerRef={[Function]}
                                                                  onScroll={[Function]}
                                                                  scrollY={true}
                                                                >
                                                                  <Scroller
                                                                    className="emotion-10"
                                                                    containerRef={[Function]}
                                                                    onScroll={[Function]}
                                                                    scrollY={true}
                                                                  >
                                                                    <Themed(OverflowContainer)
                                                                      className="emotion-10"
                                                                      containerRef={[Function]}
                                                                      onScroll={[Function]}
                                                                      scrollY={true}
                                                                    >
                                                                      <ThemeProvider>
                                                                        <ThemeProvider>
                                                                          <OverflowContainer
                                                                            className="emotion-10"
                                                                            containerRef={[Function]}
                                                                            onScroll={[Function]}
                                                                            scrollY={true}
                                                                          >
                                                                            <OverflowContainer
                                                                              className="emotion-10"
                                                                              containerRef={[Function]}
                                                                              innerRef={[Function]}
                                                                              onScroll={[Function]}
                                                                              scrollY={true}
                                                                            >
                                                                              <div
                                                                                className="emotion-9"
                                                                                onScroll={[Function]}
                                                                              >
                                                                                <EventListener
                                                                                  listeners={
                                                                                    Array [
                                                                                      Object {
                                                                                        "event": "resize",
                                                                                        "handler": [Function],
                                                                                        "target": "window",
                                                                                      },
                                                                                    ]
                                                                                  }
                                                                                />
                                                                              </div>
                                                                            </OverflowContainer>
                                                                          </OverflowContainer>
                                                                        </ThemeProvider>
                                                                      </ThemeProvider>
                                                                    </Themed(OverflowContainer)>
                                                                  </Scroller>
                                                                </Scroller>
                                                              </div>
                                                            </OverflowContainerWithShadows>
                                                          </OverflowContainerWithShadows>
                                                        </ThemeProvider>
                                                      </ThemeProvider>
                                                    </Themed(OverflowContainerWithShadows)>
                                                  </WithTheme(Themed(OverflowContainerWithShadows))>
                                                </Styled(WithTheme(Themed(OverflowContainerWithShadows)))>
                                              </withProps(Styled(WithTheme(Themed(OverflowContainerWithShadows))))>
                                            </div>
                                          </Styled(div)>
                                        </DialogHeader>
                                      </DialogBody>
                                    </DialogBody>
                                    <DialogFooter>
                                      <withProps(DialogFooter)>
                                        <DialogFooter
                                          element="footer"
                                        >
                                          <DialogHeader
                                            className="emotion-30"
                                            element="footer"
                                          >
                                            <Styled(footer)
                                              className="emotion-30"
                                              element="footer"
                                            >
                                              <footer
                                                className="emotion-29"
                                              >
                                                <DialogActions>
                                                  <DialogActions>
                                                    <div
                                                      className="emotion-28"
                                                    >
                                                      <Button
                                                        element="button"
                                                        key="0"
                                                        minimal={true}
                                                        primary={false}
                                                        size="medium"
                                                        type="button"
                                                      >
                                                        <Button
                                                          element="button"
                                                          minimal={true}
                                                          primary={false}
                                                          size="medium"
                                                          text="Cancel"
                                                          type="button"
                                                        >
                                                          <button
                                                            className="emotion-24"
                                                            type="button"
                                                          >
                                                            <Styled(span)>
                                                              <span
                                                                className="emotion-23"
                                                              >
                                                                <Styled(span)
                                                                  size="medium"
                                                                >
                                                                  <span
                                                                    className="emotion-22"
                                                                  >
                                                                    Cancel
                                                                  </span>
                                                                </Styled(span)>
                                                              </span>
                                                            </Styled(span)>
                                                          </button>
                                                        </Button>
                                                      </Button>
                                                      <Button
                                                        element="button"
                                                        key="1"
                                                        minimal={false}
                                                        primary={true}
                                                        size="medium"
                                                        type="button"
                                                      >
                                                        <Button
                                                          element="button"
                                                          minimal={false}
                                                          primary={true}
                                                          size="medium"
                                                          text="Action"
                                                          type="button"
                                                        >
                                                          <button
                                                            className="emotion-27"
                                                            type="button"
                                                          >
                                                            <Styled(span)>
                                                              <span
                                                                className="emotion-23"
                                                              >
                                                                <Styled(span)
                                                                  size="medium"
                                                                >
                                                                  <span
                                                                    className="emotion-22"
                                                                  >
                                                                    Action
                                                                  </span>
                                                                </Styled(span)>
                                                              </span>
                                                            </Styled(span)>
                                                          </button>
                                                        </Button>
                                                      </Button>
                                                    </div>
                                                  </DialogActions>
                                                </DialogActions>
                                              </footer>
                                            </Styled(footer)>
                                          </DialogHeader>
                                        </DialogFooter>
                                      </withProps(DialogFooter)>
                                    </DialogFooter>
                                  </div>
                                </DialogContent>
                              </div>
                            </Styled(div)>
                          </div>
                        </Dialog>
                      </div>
                    </FocusTrap>
                  </div>
                </Animate>
              </Transition>
            </Component>
          </WithTheme(Component)>
        </DemoDialog>
      </DemoDialog>
    </withProps(DemoDialog)>
  </div>
</Styled(div)>
`;
